rust/rsc/src/common/database.rs

```diff
@@ -157,7 +190,226 @@ pub async fn read_dbonly_blob_store(
 // ----------            Update            ----------
 
 // ----------            Delete            ----------
-pub async fn delete_all_jobs(db: &DatabaseConnection) -> Result<DeleteResult, DbErr> {
+pub async fn delete_all_jobs<T: ConnectionTrait>(db: &T) -> Result<DeleteResult, DbErr> {
     tracing::info!("Deleting ALL jobs");
     job::Entity::delete_many().exec(db).await
 }
+
+// --------------------------------------------------
+// ----------         Visible File         ----------
+// --------------------------------------------------
+
+// ----------            Create            ----------
+pub async fn create_many_visible_files<T: ConnectionTrait>(
+    db: &T,
+    visible_files: Vec<visible_file::ActiveModel>,
+) -> Result<(), DbErr> {
+    if visible_files.len() == 0 {
+        return Ok(());
+    }
+
+    let chunked: Vec<Vec<visible_file::ActiveModel>> = visible_files
+        .into_iter()
+        .chunks(MAX_SQLX_PARAMS / 5)
+        .into_iter()
+        .map(|chunk| chunk.collect())
+        .collect();
+
+    for chunk in chunked {
+        VisibleFile::insert_many(chunk).exec(db).await?;
```

Comment:
Is there a benefit to awaiting after all chunks inserts are made?

Reply:
Yep, but I'd like to push that off into a separate "join all futures" PR